samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -65,15 +73,19 @@
    * Key is chosen to be processorId since at a time only one placement action can be in progress on a container.
    */
   private final ConcurrentHashMap<String, ContainerPlacementMetadata> actions;
+  private final BoundedFifoCache<UUID> dequeuedActionsCache;
```

Comment:
Could be called placementRequestsCache?

/**
In-memory cache of placement requests dequeued from the coordinator-store, used to dedup requests with the same request UUID.
*/

Reply:
Sure added docs